### PR TITLE
[SPARK-47726][DOC] Document push-based shuffle metrics

### PR DESCRIPTION
**Pull request by LucaCanali**
### What changes were proposed in this pull request?
This adds documentation for the push-based shuffle metrics

### Why are the changes needed?
The push-based shuffle metrics are currently not documented

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
N/A

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: docs/monitoring.md [hunk 1]
```diff
@@ -1291,6 +1291,17 @@ These metrics are exposed by Spark executors.
   - shuffleRemoteBytesReadToDisk.count
   - shuffleTotalBytesRead.count
   - shuffleWriteTime.count
+  - Metrics related to push-based shuffle:
```
Comment (dongjoon-hyun) [inline docs/monitoring.md:1294]: Are these always exposed when we don't use `push-based shuffle`?

Comment (LucaCanali) [reply docs/monitoring.md:1294]: Yes, they are. I understand https://issues.apache.org/jira/browse/SPARK-42203 is supposed to take care of that issue.

Comment (dongjoon-hyun) [reply docs/monitoring.md:1294]: Thanks! I pinged on SPARK-42203. Let's wait for a few days.

**Comment by dongjoon-hyun** [comment]
Sorry for making you wait. It seems that SPARK-42203 will not happen. Let's proceed this first.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @LucaCanali .
Merged to master for Apache Spark 4.0.0.

**Comment by LucaCanali** [comment]
Thank you @dongjoon-hyun

**Pull request closed without merging**